Document font-palette & @font-palette-values with its descriptors (#22519)
mdn/content: files/en-us/web/css/@font-palette-values/font-family/index.md

---
title: font-family
slug: Web/CSS/@font-palette-values/font-family
page-type: css-at-rule-descriptor
tags:
  - CSS
  - CSS Fonts
  - Color Fonts
  - CSS Descriptor
  - Reference
browser-compat: css.at-rules.font-palette-values.font-family
---

{{CSSRef}}

The [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) **`font-family`** is used to specify which font-family palette values are to be applied to. This need to match exactly the values used when setting the CSS [font-family](/en-US/docs/Web/CSS/font-family).

## Syntax

```css
@font-palette-values --Dark-mode {
  font-family: "Bungee Spice";
  /* other palette settings follow */
}
```

> The `font-family` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) means that the settings that follow apply to the specified font-family. Other [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) can be set for other font-families using the same [&lt;dashed-ident&gt;s](/en-US/docs/Web/CSS/dashed-ident), which means if you have multiple Color Fonts you can use the same identifier for each.

Author: Cool I shall work on this

> The `font-family` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) means that the settings that follow apply to the specified font-family. Other [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) can be set for other font-families using the same [&lt;dashed-ident&gt;s](/en-US/docs/Web/CSS/dashed-ident), which means if you have multiple Color Fonts you can use the same identifier for each.

Author: again there should be a dash in "font family"

> The `font-family` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) means that the settings that follow apply to the specified font-family. Other [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) can be set for other font-families using the same [&lt;dashed-ident&gt;s](/en-US/docs/Web/CSS/dashed-ident), which means if you have multiple Color Fonts you can use the same identifier for each.

Author: This is now done shall update the PR

### Values

- `<family-name>`

>   - : Specifies the name of the font family.

Author: I feel this should have a dash for reasons: https://github.com/mdn/content/pull/22519#discussion_r1039724999

>   - : Specifies the name of the font family.

Author: This is now done shall update the PR

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

### Naming syntax

> As an example when `font-family` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) is used in the [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) the value is identical in each instance, eg `"Bungee Spice"`.

Author: In this example, when the `font-family` descriptor is used in the [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) at-rule, the same value is used for the `font-family`, when it is declared.

> As an example when `font-family` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) is used in the [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) the value is identical in each instance, eg `"Bungee Spice"`.

Author: This is now done shall update the PR

#### HTML

```html
<h2>This is spicy</h2>
<h2 class="extra-spicy">This is extra hot & spicy</h2>
```

#### CSS

```css
@import url(https://fonts.googleapis.com/css2?family=Bungee+Spice);
@font-palette-values --bungee-extra-spicy {
  font-family: "Bungee Spice";

>   override-colors: 

Author: No this is only needed when using an alternate base-palette is used and not all color-fonts have more than 1 base-palette

    0 DarkRed,
    1 Red;
}
h2 {
  font-family: "Bungee Spice";
}
h2.extra-spicy {
  font-palette: --bungee-extra-spicy;
}
```

#### Result

> {{EmbedLiveSample("Naming syntax")}}

Author: See here: https://github.com/mdn/content/pull/22519#discussion_r1036381117

> {{EmbedLiveSample("Naming syntax")}}

Author: This is now done shall update the PR

> ### Using the same identifier for multiple font-families example

Author: This is now done shall update the PR

> In this example we see the [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) set for 2 font-families and the [&lt;dashed-ident&gt;](/en-US/docs/Web/CSS/dashed-ident) is the same for each font-family. When setting the [font-palette](/en-US/docs/Web/CSS/font-palette) it can be done for all the elements at the same time. This may be done so that you can change the Color Fonts colors to match your sites branding.

Reviewer: This is a good example of a description for the example 🙂! 

```suggestion
In this example, two [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) at-rules are set for two font families, but both the at-rules use the same dashed-ident identifier, `--Dark Mode`. This helps to set the [font-palette](/en-US/docs/Web/CSS/font-palette) property for multiple elements, `h1` and `h2` in this case, at the same time. This can be useful when you want to update font colors to match your site's branding.
```

```css
@font-palette-values --Dark-Mode {
  font-family: "Bungee Spice";
  /* palette settings for Bungee Spice */
}
@font-palette-values --Dark-Mode {
  font-family: Bixa;
  /* palette settings for Bixa */
}
h1, h2 {
  font-palette: --Dark-Mode;
}
h1 {
  font-family: "Bungee Spice";
}
h2 {
  font-family: Bixa;
}
```

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

- {{cssxref("@font-face/font-family", "font-family")}}
- {{cssxref("@font-palette-values/", "@font-palette-values")}}

> - {{cssxref("@font-palette-values/font-family", "font-family")}}
> - {{cssxref("@font-palette-values/override-colors", "override-colors")}}
> - {{cssxref("font-palette/", "font-palette")}}

Reviewer: ```suggestion
- {{cssxref("@font-palette-values/font-family", "font-family")}} descriptor
- {{cssxref("@font-palette-values/override-colors", "override-colors")}} descriptor
- {{cssxref("font-palette/", "font-palette")}} property
```

### Values

- `<family-name>`
  - : Specifies the name of the font family.

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

### Using matching family names

As an example when `font-family` [descriptor](/en-US/docs/Glossary/Descriptor_(CSS)) is used in the [@font-palette-values](/en-US/docs/Web/CSS/@font-palette-values) the value is identical in each instance, eg `"Bungee Spice"`.

#### HTML

```html
<h2>This is spicy</h2>
<h2 class="extra-spicy">This is extra hot & spicy</h2>
```

#### CSS

```css
@import url(https://fonts.googleapis.com/css2?family=Bungee+Spice);
@font-palette-values --bungee-extra-spicy {
  font-family: "Bungee Spice";
  override-colors: 
    0 DarkRed,
    1 Red;
}

h2 {
  font-family: "Bungee Spice";
}

h2.extra-spicy {
  font-palette: --bungee-extra-spicy;
}
```

#### Result

> {{EmbedLiveSample("Naming syntax")}}

Reviewer: to match the example's title:

```suggestion
{{EmbedLiveSample("Using_matching_family_names")}}
```

Author: This is now done
